### PR TITLE
AI use beacon to inform human allies on ground rush

### PR DESCRIPTION
**Pull request by dnqbob**
Life improvement on combat with AI.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs [hunk 1]
```diff
@@ -57,6 +57,9 @@ public void Tick(Squad owner)
 					owner.Bot.QueueOrder(new Order("AttackMove", u, Target.FromCell(owner.World, owner.TargetActor.Location), false));
 
 				// We have gathered sufficient units. Attack the nearest enemy unit.
+				// Inform human allies about AI's rush attack.
+				owner.Bot.QueueOrder(new Order("PlaceBeacon", owner.SquadManager.Player.PlayerActor, Target.FromCell(owner.World, owner.TargetActor.Location), false)
+				{ SuppressVisualFeedback = true });
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs]: Style nit: This needs one level of indentation.

**Review by abcdefg30 — COMMENTED** [review]
Do we want to have the bot send a message alongside? Just a beacon might be confusing at first.

**Comment by pchote** [comment]
How often does this trigger in practice? I can imagine that players would get frustrated if they are being pinged every minute or more.

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs (hunk 1 above)
Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs:62]: > Do we want to have the bot send a message alongside? Just a beacon might be confusing at first.

I want to but I can't find the code on how to chat in team

**Comment by dnqbob** [comment]
Yeah it is a problem when mod set ~~attack interval~~ "SquadSize" too small or "RushInterval" too short.

but at default "SquadSize" is 20, which means AI need at least two minutes to build, and "SquadSizeRandomBonus" also make squad bigger.

**Comment by dnqbob** [comment]
> Do we want to have the bot send a message alongside? Just a beacon might be confusing at first.

I want to but I can't find the code on how to chat in team.

update: and I also need to make the code here as simple as possible for I am afraid of team chat will make the code here too complicated.

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs (hunk 1 above)
Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs]: ok,  a table  should enough?

**Comment by dnqbob** [comment]
As for AI chat, I believe a new bot module should be in OpenRA for that. which will let AI not only chat in team but also chat in world for taunt or greeting.

**Pull request closed without merging**